.gitignore

```diff
@@ -1,2 +1,5 @@
 html/
 node_modules/
+venv/
+__pycache__/
+build/
```

Comment:
Consider adding `.mypy_cache/` for those who use mypy, and `.vscode/` for those who use vscode.

Reply:
See comment RE: MyPy
It's previously been contributors' responsibility to keep their IDE files out of source control (ie with a user gitignore) and changing that precedent is out of scope of this PR.